crates/wasi-component-loader/src/resources/kafka_producer/murmur2.rs

```diff
@@ -0,0 +1,77 @@
+/// Murmur2 hash implementation matching Java Kafka client
+/// This ensures messages with the same key go to the same partition
+/// across different Kafka client implementations
+pub(super) fn hash(data: &[u8]) -> u32 {
```

Comment:
This hash function is tiny. Let's for once not add one more dependency from six years ago without a maintainer to this project, no?